packages/babel-plugin/src/plugin/index.ts

```diff
@@ -99,6 +112,50 @@ function getRealmTypeForTSTypeReference(path: NodePath<types.TSTypeReference>):
     return { type: "dictionary", objectType: objectType?.type, optional: objectType?.optional };
   } else if (isRealmTypeAlias(path, "Mixed") || isRealmTypeAlias(path, "Mixed", null)) {
     return { type: "mixed" };
+  } else if (isRealmTypeAlias(path, "LinkingObjects")) {
+    const classPropertyNode = path.parentPath?.parentPath?.node;
+    if (
+      // Keep TS happy
+      !types.isClassProperty(classPropertyNode) ||
+      classPropertyNode.optional
+    ) {
+      throw new Error(getLinkingObjectsError("Properties of type LinkingObjects cannot be optional"));
```

Comment:
🥇 for preconditions + errors ... Perhaps we're missing some elsewhere?

Reply:
Yeah probably – I actually meant to ask your opinion on whether this is overkill or not? i.e. should we just rely on the type definitions? 

I think throwing an explicit error is probably better though, as sometimes people will ignore TS errors

Reply:
Would it be possible to send this to babel somehow, in a way that wouldn't make it completely unusable if a property is misconfigured, but simply error through to the bundler. I don't know what's available for babel plugins to report errors or warnings.

Reply:
Yeah that is a good idea – I'll do some research.

Reply:
Actually I think this should be a `throw`, as if you replace it with e.g. a `console.warn`, you then end up with `Property 'xxx' of type 'linking objects' cannot be nullable.` – so you are specifying an invalid schema.

However if you use `throw path.buildCodeFrameError(errorMsg)`, you get a better error message which shows the source of the error, so I will switch to that.

![image](https://user-images.githubusercontent.com/5458070/193617006-d375a1b2-38f8-46f7-a612-ce2ff5cdbf96.png)

Reply:
I couldn't really think of any other good preconditions. The type argument for List/Dictionary/Set is optional and if they supply more than 1 type argument for some reason it will still work as intended... doesn't really seem worth checking. Can you think of anywhere else?